service/credential/credential.go

```diff
@@ -42,11 +44,21 @@ func GetAzureConfig(k8sClient kubernetes.Interface, name string, namespace strin
 		return nil, microerror.Mask(err)
 	}
 
+	partnerID, err := valueFromSecret(credential, PartnerIDKey)
+	if err != nil {
+		// No having partnerID in the secret means that customer has not
+		// upgraded yet to use the Azure Partner Program. In that case we set a
+		// constant random generated GUID that we haven't registered with Azure.
+		// When all customers have migrated, we should error out instead.
+		partnerID = defaultAzureGUID
+	}
```

Comment:
Instead of erroring out when the value is not present, I'm using a default GUID that's not registered with Azure. That way this is not a breaking change anymore, and customers can start using these changes on the Azure operator even before getting a valid GUID registered on the Azure Partners Program.

Reply:
Just to be sure: When GUIDs actually have the group part of the ID, does this match with valid ones? At least this needs to be tested so that Azure doesn't flip with it :slightly_smiling_face: